src/modules/Channel/components/MessageFeedbackModal/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { ReactElement, useContext, useRef } from 'react';
+import { LocalizationContext } from '../../../../lib/LocalizationContext';
+import Modal from '../../../../ui/Modal';
+import Button, { ButtonTypes } from '../../../../ui/Button';
+import Input from '../../../../ui/Input';
+import Label, { LabelColors, LabelTypography } from '../../../../ui/Label';
+import './index.scss';
+import { useMediaQueryContext } from '../../../../lib/MediaQueryContext';
+import { CoreMessageType } from '../../../../utils';
+import { FeedbackRating } from '@sendbird/chat/message';
+import { useKeyDownV2 } from './hooks/useKeyDownV2';
+
+export interface MessageFeedbackModalProps {
+  selectedFeedback: FeedbackRating;
+  message: CoreMessageType;
+  onCancel?: () => void;
+  onSubmit?: (comment: string) => void;
+  onRemove?: () => void;
+}
+
+export default function MessageFeedbackModal(props: MessageFeedbackModalProps): ReactElement {
+  const {
+    selectedFeedback,
+    message,
+    onCancel,
+    onSubmit,
+    onRemove,
+  } = props;
+
+  const { stringSet } = useContext(LocalizationContext);
+  const { isMobile } = useMediaQueryContext();
+
+  const modalRef = useRef(null);
+  const inputRef = useRef(null);
+  const onKeyDown = useKeyDownV2({
+    ref: modalRef,
+    keyDownCallbackMap: {
+      Enter: () => onSubmit?.(inputRef.current.value ?? ''),
+      Escape: () => onCancel?.(),
+    },
+  });
+
+  return (
+    <div onKeyDown={onKeyDown}>
+      <Modal
+        contentClassName='sendbird-message-feedback-modal-content__mobile'
+        type={ButtonTypes.PRIMARY}
+        onCancel={onCancel}
+        onSubmit={() => {
+          onSubmit?.(inputRef.current.value ?? '');
+        }}
+        submitText={stringSet.BUTTON__SUBMIT}
+        renderHeader={() => (
+          <div className='sendbird-modal__header'>
+            <Label
+              type={LabelTypography.H_1}
+              color={LabelColors.ONBACKGROUND_1}
+              className='sendbird-message-feedback-modal-header'
+            >
+              {stringSet.FEEDBACK_MODAL_TITLE}
+            </Label>
+          </div>
+        )}
+        customFooter={
+          <div className='sendbird-message-feedback-modal-footer__root'>
+            {
+              !isMobile && message?.myFeedback && selectedFeedback === message.myFeedback.rating
+                ? <Button type={ButtonTypes.WARNING} onClick={onRemove}>
+                  <Label type={LabelTypography.BUTTON_3} color={LabelColors.ERROR}>
+                    {stringSet.BUTTON__REMOVE_FEEDBACK}
+                  </Label>
+                </Button>
+                : <div/>
```

Comment:
It looks a bit awkward that a closing tag is next to the `:` ..?

Reply:
Ah its an empty div: <div></div>. I tried with <></> but its not an element so breaks the intended flex layout. So I had to use empty div instead.

Reply:
Then 
```
!isMobile && message?.myFeedback && selectedFeedback === message.myFeedback.rating
                ? <Button type={ButtonTypes.WARNING} onClick={onRemove}>
                  <Label type={LabelTypography.BUTTON_3} color={LabelColors.ERROR}>
                    {stringSet.BUTTON__REMOVE_FEEDBACK}
                  </Label>
                </Button>
```
^ this block won't have a closing div tag.

How about this? Even it has a little duplicated class names, but it looks more natural I think.  
```
customFooter={
// To display the delete feedback button on the wide screen
!isMobile && message?.myFeedback && selectedFeedback === message.myFeedback.rating
  ? <div className='sendbird-message-feedback-modal-footer__root'>...</div>
  : <div className='sendbird-message-feedback-modal-footer__root' />
```

Reply:
if we were to take out 
```
!isMobile && message?.myFeedback && selectedFeedback === message.myFeedback.rating
```
outside of
```
<div className='sendbird-message-feedback-modal-footer__root'>
```
then we need to have
```
<div className='sendbird-message-feedback-modal-footer__right-content'>
```
in both web and mobile `<div>` which creates duplicated code. Are we ok with this or did I miss something?

Reply:
Oh sorry for the delay. Yeah I saw the full code and now I got your point. Hmm it's weird that we need `<div />` not to break the layout. We probably need to see the style setting.